### ISSUE TITLE
Add onBlur and onFocus to AutoCompolete and DatePicker component

### ISSUE THREAD

**Issue by JozefBiros** [ant-design/ant-design#7263]
### What problem does this feature solve?
I am using ant design with redux-form library. Redux-form is using methods onBlur and onFocus for sync validation and in AutoComplete component there is no onBlur and onFocus available.

As I understood it correctly AutoComplete is extending Select component which has onFocus and onBlur methods, so there is no valid reason for AutoComplete to not have it.

I had have same issue with DatePicker which is only input, but doesnt have onBlur and onFocus methods.

### What does the proposed API look like?
Add onBlur and onFocus method to all components which supports it.

**Comment by afc163** [comment]
> using ant design with redux-form library

I think it will be great if antd could be compatiable with redux-form, could you help us and try to propose PR for this?

**Comment by JozefBiros** [comment]
Sure I would love to, I can show you how I ensure compatibility on basic Ant Input and if you like my way I can do other data entry components. I just have to figure out how I do PR I never did one.

**Comment by ndussart** [comment]
I didn't check the AutoComplete component but for the DatePicker component I mapped the onOpenChange prop with the redux-form input prop like this : 

`onOpenChange={status => status ? input.onFocus() : input.onBlur()}`

**Comment by JozefBiros** [comment]
Yes, I solved DatePicker same way as you and AutoComplete can take child component Input so I passed onFocus() and onBlur() there. But it is not very elegenat because DatePicker in its core is just simple Input.

**Comment by chhuang** [comment]
Does this help? https://github.com/zhdmitry/redux-form-antd

**Comment by rvanmarkus** [comment]
I also really like to see an onFocus / onBlur on the autocomplete API. I need it to know when the user has selected the autocomplete field so I can get some async options based on the value of an older field.
